src/app/layout/notification-bar/dynamic-notifications/dynamic-notification.interface.ts

```diff
@@ -0,0 +1,5 @@
+export interface DynamicNotification {
+
+    componentInputs?: { [key: string]: any };
```

Comment:
Not necessary, return an instance like `ModalService` does, and set properties on the instance.
This approach makes code non-refactorable and untraceable.